pyrokinetics/pyro.py

```diff
@@ -171,21 +171,35 @@ def read_gk_file(self, gk_file=None, gk_code=None):
         if gk_file is not None:
             self.gk_file = gk_file
 
-        if gk_code is not None:
+        if self.gk_file is None:
+            raise ValueError("Please specify gk_file")
+
+        # if gk_code is not given, try inferring from possible GKCodes
+        if gk_code is None:
+            for key, gk in gk_codes.items():
+                try:
+                    gk.verify(self.gk_file)
+                    self.gk_code = key
+                    break
+                except Exception:
+                    continue
+        else:
             self.gk_code = gk_code
 
-        if self.gk_code is None or self.gk_file is None:
-            raise ValueError("Please specify gk_code and gk_file")
-        else:
+        # if self.gk_code is still None, we couldn't infer a file type
+        # Must also check against GKCode, and due to inheritance we need
+        # to check type directly rather than using isinstance
+        if self.gk_code is None or type(self.gk_code) == GKCode:
+            raise ValueError("Could not determine gk_code from file type")
 
-            # If equilibrium already loaded then it won't load the input file
-            if hasattr(self, "eq"):
-                template = True
-            else:
-                template = False
+        # If equilibrium already loaded then it won't load the input file
+        if hasattr(self, "eq"):
+            template = True
+        else:
+            template = False
```

Comment:
If we're touching this:

```suggestion
        # If equilibrium already loaded then it won't load the input file
        template = hasattr(self, "eq"):
```